### ISSUE TITLE
Dialer.DialPartition ignores address parameter

### ISSUE THREAD

**Issue by robfig** [segmentio/kafka-go#397]
The `address` parameter is ignored in Dialer.DialPartition, since the address is provided in the Partition struct. I think it is confusing and should be removed. 
https://github.com/segmentio/kafka-go/blob/master/dialer.go#L140

**Comment by achille-roussel** [comment]
You're right, but since this will be a breaking change it's risky to make it.

Would you be available to contribute an update to the documentation?

**Comment by sammy-hughes** [comment]
Couldn't the parameter name instead be `_`'ed?
